### ISSUE TITLE
Suffix items

### ISSUE THREAD

**Issue by pombredanne** [kmike/datrie#25]
The suffix method by @xmonader is great but it is missing returning items or values. It should instead instead of just returning keys.

**Comment by superbobry** [comment]
Do you mean you want `suffix_items` instead of `suffixes`?

**Comment by pombredanne** [comment]
@superbobry exactly. Eventually with the iterator variants, in a way that's symmetric to the `prefix*` functions

**Comment by pombredanne** [comment]
Sorry, I see now that I had written mumbo jumbo in my initial post...

**Comment by superbobry** [comment]
Right. I think we should go this way and add `suffix_items` and `suffix_values`.

And while we're on it, why not revise the current prefix/suffix API? For instance, I think we should deprecate (and later drop) non-iterator methods (e.g. `prefixes`) in favour of `iter_prefixes` & friends. There's no point in keeping them, since you can always wrap the `iter_*` call into a `list`. Once we do this, I suggest to go further and remove the `iter_*` prefix to make the API is more Python3-ish.

@kmike what do you think?

**Comment by pombredanne** [comment]
:+1:

**Comment by kmike** [comment]
In Python 3 `keys()` and alike methods don't just return iterators; they return key views which e.g. support membership checks, `__repr__` and `__str__` method which show values, len() method. AFAIK for custom objects you need to implement this logic yourselves; without it I found lists more convenient than iterators. That's why I went with non-iterator methods in first place - it was less work, more convenience for explorations in console, and there are iter_ methods for those who need memory-efficient production code.

**Comment by superbobry** [comment]
This is a valid point. Anyway, I can sketch the view-based version of `datrie` in a separate branch and then we'll see if there's any benefit.
